File: leetcode/js/49. Group Anagrams.js
function groupAnagrams(strs){
    let result = []
    let helper = {}
    for(let s of strs){
      let tmpChar = s.split("")
      tmpChar.sort()
      let tmp = tmpChar.join("")
      if(Object.keys(helper).includes(tmp)){
         result[helper[tmp]].push(s)
         continue;
      }
      let curr =[]
      curr.push(s)
      result.push(curr)
      helper[tmp] = result.length - 1 
    }
    return result
  }
  
  const dataSet = ["eat","tea","tan","ate","nat","bat"]
  
  console.log(groupAnagrams(dataSet))

File: leetcode/python/49. Group Anagrams.py
def groupAnagrams(strs):
  result = []
  helper = {}
  for s in strs:
    tmpChar = s.split()
    sorted(tmpChar)
    tmp = "".join(tmpChar)
    if helper in list(helper.keys()):
      result[helper[tmp]].append(s)
      continue
    curr = []
    curr.append(s)
    result.append(curr)
    helper[tmp] = len(result) - 1
  return result

dataSet = ["eat","tea","tan","ate","nat","bat"]

print(groupAnagrams(dataSet))